akka-cluster/src/main/scala/akka/cluster/ClusterDaemon.scala

```diff
@@ -992,6 +1034,15 @@ private[cluster] class ClusterCoreDaemon(publisher: ActorRef) extends Actor with
       }
       val newGossip = localGossip copy (members = newMembers, overview = newOverview, version = newVersion)
 
+      if (!exitingInProgress && newGossip.member(selfUniqueAddress).status == Exiting) {
+        // Leader is moving itself from Leaving to Exiting.
+        // ExitingCompleted will be received via CoordinatedShutdown to continue
+        // the leaving process. Meanwhile the gossip state is not marked as seen.
+        exitingInProgress = true
+        logInfo("Exiting (leader), starting coordinated shutdown")
+        coordShutdown.run()
```

Comment:
I have user phases in mind and also the typical "stop incoming http requests first". I will add those to the pre-defined configured phases, then tasks can be added in akka-http and user code.

I think a management api would simply call `CoordinatedShutdown(system).run()` and then it will just work - I hope :)